extensions/amp-story/1.0/amp-story.js

```diff
@@ -1322,6 +1330,32 @@ export class AmpStory extends AMP.BaseElement {
     this.activePage_.setState(pageState);
   }
 
+  /**
+   * Reacts to sidebar state updates.
+   * @param {boolean} sidebarState
+   * @private
+   */
+  onSidebarStateUpdate_(sidebarState) {
+    const actions = Services.actionServiceForDoc(this.getAmpDoc());
+    if (this.win.MutationObserver) {
+      sidebarObserver = new this.win.MutationObserver(() => {
+        this.storeService_.dispatch(Action.TOGGLE_SIDEBAR,
+            this.sidebar_.hasAttribute('open'));
+      });
+      if (sidebarState) {
+        sidebarObserver.observe(this.sidebar_, {attributes: open});
+      } else {
+        sidebarObserver.disconnect();
```

Comment:
Because you're instantiating a new `MutationObserver` every single time `onSidebarStateUpdate_` is called, here you're calling `disconnect` on a new observer, and not on the one you actually wanted to disconnect.

Reply:
Ahh I see. Like I said in this comment, I couldn't figure out a way to ensure the same observer would be used because when i tried to store the observer as a field, the pause state wouldn't work and through debugging, i would notice that the disconnect method would be called immediately. I still can't figure why the disconnect is being called. I'll keep trying, but if you a suggestion for a possible reason/work around please let me know :)

Here is the comment btw: https://github.com/ampproject/amphtml/pull/17836#discussion_r219222847

Reply:
Oh, sorry, I misunderstood your previous comment.

So the issue here, is that you're instantiating a new `MutationObserver` every time the callback is called.
What you want, is have only one that you re use all the time, so you can just start observing when the sidebar is open, and disconnect when the sidebar is closed.

Pseudo code:

```
constructor() {
  this.sidebarObserver_ = null;
}

onSidebarStateUpdate_(sidebarState) {
  if (this.win.MutationObserver) {
    // If this.sidebarObserver_ exists, do nothing, and if it doesn't, instantiate it
    // If we're opening the sidebar, this.sidebarObserver.observe(/** arguments */);
    // If we're closing the sidebar, this.sidebarObserver.disconnect();
  } else {
    // If mutation observer is not supported and we're opening the sidebar, unpause the story since there's no way to know when the sidebar will actually be closed
  }
}
```

Let me know if you have any question :)